backend/inspirehep/search/api.py

```diff
@@ -192,15 +225,21 @@ def query_by_user_role(self, query_string):
         fulltext_highlight_options = dict(
             fragment_size=160, type="fvh", number_of_fragments=1, order="score"
         )
+        query = IQ(query_string, self)
+
+        if "citedby" in query_string:
+            query = query.to_dict()
+            self.replace_recid_in_citedby_query(query)
```

Comment:
Why we have this logic inside the [execute](https://github.com/inspirehep/inspirehep/pull/2674/files#diff-19b96149fef1fdaddf2df9e639f18cc6b2fd728e0d2f7a6e5feb1f0514fece0cL179)?

Reply:
cause we parse the query here and I need a parsed query to replace recid